samples/client/petstore/swift5/vaporLibrary/Sources/PetstoreClient/Extensions.swift

```diff
@@ -0,0 +1,134 @@
+        var arrayContainer = nestedUnkeyedContainer(forKey: key)
+        try arrayContainer.encode(contentsOf: values)
+    }
+
+    public mutating func encodeArrayIfPresent<T>(_ values: [T]?, forKey key: Self.Key) throws where T: Encodable {
+        if let values = values {
+            try encodeArray(values, forKey: key)
+        }
+    }
+
+    public mutating func encodeMap<T>(_ pairs: [Self.Key: T]) throws where T: Encodable {
+        for (key, value) in pairs {
+            try encode(value, forKey: key)
+        }
+    }
+
+    public mutating func encodeMapIfPresent<T>(_ pairs: [Self.Key: T]?) throws where T: Encodable {
+        if let pairs = pairs {
+            try encodeMap(pairs)
+        }
+    }
+
+}
+
+extension KeyedDecodingContainerProtocol {
+
+    public func decodeArray<T>(_ type: T.Type, forKey key: Self.Key) throws -> [T] where T: Decodable {
+        var tmpArray = [T]()
+
+        var nestedContainer = try nestedUnkeyedContainer(forKey: key)
+        while !nestedContainer.isAtEnd {
+            let arrayValue = try nestedContainer.decode(T.self)
+            tmpArray.append(arrayValue)
+        }
+
+        return tmpArray
+    }
+
+    public func decodeArrayIfPresent<T>(_ type: T.Type, forKey key: Self.Key) throws -> [T]? where T: Decodable {
+        var tmpArray: [T]?
+
+        if contains(key) {
+            tmpArray = try decodeArray(T.self, forKey: key)
+        }
+
+        return tmpArray
+    }
+
+    public func decodeMap<T>(_ type: T.Type, excludedKeys: Set<Self.Key>) throws -> [Self.Key: T] where T: Decodable {
+        var map: [Self.Key: T] = [:]
+
+        for key in allKeys {
+            if !excludedKeys.contains(key) {
+                let value = try decode(T.self, forKey: key)
+                map[key] = value
+            }
+        }
+
+        return map
+    }
+
+}
+
+extension UUID: Content { }
+
+extension URL: Content { }
+
+extension Bool: Content { }
+
+extension Set: ResponseEncodable where Element: Content {
+    public func encodeResponse(for request: Vapor.Request) -> EventLoopFuture<Vapor.Response> {
+        let response = Vapor.Response()
+        do {
+            try response.content.encode(Array(self))
+        } catch {
+            return request.eventLoop.makeFailedFuture(error)
+        }
+        return request.eventLoop.makeSucceededFuture(response)
+    }
+}
+
+extension Set: RequestDecodable where Element: Content {
+    public static func decodeRequest(_ request: Vapor.Request) -> EventLoopFuture<Self> {
+        do {
+            let content = try request.content.decode([Element].self)
+            return request.eventLoop.makeSucceededFuture(Set(content))
+        } catch {
+            return request.eventLoop.makeFailedFuture(error)
+        }
+    }
+}
+
+extension Set: Content where Element: Content { }
+
+extension AnyCodable: Content {}
+extension AnyEncodable: Content {}
+extension AnyDecodable: Content {}
```

Comment:
This line is creating the following build error
`Extension outside of file declaring struct 'AnyDecodable' prevents automatic synthesis of 'encode(to:)' for protocol 'Encodable'`